### PR TITLE
Generate error on `goal clerk group` on signed transaction

### PR DESCRIPTION
**Pull request by tsachiherman**
## Summary

Using the existing `goal clerk group` would allow you to modify a file containing a signed transactions. This, in turn would invalidate the signature on that file, making it unusable.

To address this, this PR tests to see if any of the transactions in the transaction file provided to `goal clerk group` were previously signed. If so, it would fail the grouping command with an error.

## Test Plan

Unit test added to verify solution correctness.

### CODE REVIEW

File: crypto/curve25519.go [hunk 1]
```diff
@@ -125,6 +125,14 @@ func ed25519Verify(public ed25519PublicKey, data []byte, sig ed25519Signature) b
 // produced by a holder of a cryptographic secret.
 type Signature ed25519Signature
 
+// BlankSignature is an empty signature structure, containing nothing but zeroes
+var BlankSignature = Signature{}
+
+// Blank tests to see if the given signature contains only zeros
+func (s *Signature) Blank() bool {
```
Comment (tsachiherman) [inline crypto/curve25519.go:132]: The function name "Blank" was chosen to align with the method of the same name used for Multisig and LogicSig

File: test/e2e-go/cli/goal/expect/goalExpectCommon.exp [hunk 2]
```diff
@@ -675,12 +679,18 @@ proc ::AlgorandGoal::GetHighestFundedAccountForWallet { WALLET_NAME NODE_DIR } {
         set ACCOUNT_ADDRESS "NONE"
         expect {
             timeout { ::AlgorandGoal::Abort "Timed out finding primary wallet: $WALLET_NAME"  }
+            -re {\[n/a\]\t([a-zA-Z0-9]+)\t([a-zA-Z0-9]+)\t\[n/a\]} {
+                    set CURRENT_ACCOUNT_ADDRESS $expect_out(2,string);
+                    if { $ACCOUNT_BALANCE == 0 } then { set ACCOUNT_ADDRESS $CURRENT_ACCOUNT_ADDRESS };
+                    exp_continue;
+                }
```
Comment (tsachiherman) [inline test/e2e-go/cli/goal/expect/goalExpectCommon.exp:686]: This adds the support to allow us to get the account even when the algod isn't running.

File: test/e2e-go/cli/goal/expect/goalClerkGroupTest.exp [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    set TXN_FILE1 "$TEST_PRIMARY_NODE_DIR/txns1"
+    set TRANSFER_AMOUNT1 3800
+    ::AlgorandGoal::AccountTransfer $PRIMARY_WALLET_NAME "" $PRIMARY_ACCOUNT_ADDRESS $TRANSFER_AMOUNT1 $PRIMARY_ACCOUNT_ADDRESS $FEE_AMOUNT $TEST_PRIMARY_NODE_DIR $TXN_FILE1
+
+    set TXN_FILE2 "$TEST_PRIMARY_NODE_DIR/txns2"
+    set TRANSFER_AMOUNT2 5200
+    ::AlgorandGoal::AccountTransfer $PRIMARY_WALLET_NAME "" $PRIMARY_ACCOUNT_ADDRESS $TRANSFER_AMOUNT2 $PRIMARY_ACCOUNT_ADDRESS $FEE_AMOUNT $TEST_PRIMARY_NODE_DIR $TXN_FILE2
+
+    set TXNS_FILE "$TEST_PRIMARY_NODE_DIR/txns"
+    exec cat $TXN_FILE1 > $TXNS_FILE
```
Comment (algorandskiy) [inline test/e2e-go/cli/goal/expect/goalClerkGroupTest.exp:42]: cat $TXN_FILE1 $TXN_FILE2 > $TXNS_FILE

**Review by algorandskiy — COMMENTED** [review]
Looks good. Could you also add a test that fails when attempt to group signed transaction(s)?

**Comment by tsachiherman** [comment]
> Looks good. Could you also add a test that fails when attempt to group signed transaction(s)?

Huh ? I did..

**Comment by algorandskiy** [comment]
Yeah, I got confused with ::AlgorandGoal::AssembleGroup  vs direct `goal clerk group` call

**Review by algorandskiy — APPROVED** [review]

**Pull request merged**